src/sentry/incidents/subscription_processor.py

```diff
@@ -126,98 +154,197 @@ def process_update(self, subscription_update):
         # before the next one then we might alert twice.
         self.update_alert_rule_stats()
 
-    def trigger_alert_threshold(self):
+    def trigger_alert_threshold(self, trigger):
         """
         Called when a subscription update exceeds the value defined in the
-        `alert_rule.alert_threshold`, and there is not already an active incident. Increments the
-        count of how many times we've consecutively exceeded the threshold, and if
-        above the `threshold_period` defined in the alert rule then create an incident.
+        `trigger.alert_threshold`, and the trigger hasn't already been activated.
+        Increments the count of how many times we've consecutively exceeded the threshold, and if
+        above the `threshold_period` defined in the alert rule then mark the trigger as
+        activated, and create an incident if there isn't already one.
         :return:
         """
-        self.alert_triggers += 1
-        if self.alert_triggers >= self.alert_rule.threshold_period:
-            detected_at = to_datetime(self.last_update)
-            self.active_incident = create_incident(
-                self.alert_rule.organization,
-                IncidentType.ALERT_TRIGGERED,
-                # TODO: Include more info in name?
-                self.alert_rule.name,
-                alert_rule=self.alert_rule,
-                # TODO: Incidents need to keep track of which metric to display
-                query=self.subscription.query,
-                date_started=detected_at,
-                date_detected=detected_at,
-                projects=[self.subscription.project],
-            )
+        self.trigger_alert_counts[trigger.id] += 1
+        if self.trigger_alert_counts[trigger.id] >= self.alert_rule.threshold_period:
+            # Only create a new incident if we don't already have an active one
+            if not self.active_incident:
+                detected_at = to_datetime(self.last_update)
+                self.active_incident = create_incident(
+                    self.alert_rule.organization,
+                    IncidentType.ALERT_TRIGGERED,
+                    # TODO: Include more info in name?
+                    self.alert_rule.name,
+                    alert_rule=self.alert_rule,
+                    # TODO: Incidents need to keep track of which metric to display
+                    query=self.subscription.query,
+                    date_started=detected_at,
+                    date_detected=detected_at,
+                    projects=[self.subscription.project],
+                )
+            # Now create (or update if it already exists) the incident trigger so that
+            # we have a record of this trigger firing for this incident
+            incident_trigger = self.incident_triggers.get(trigger.id)
```

Comment:
Yeah it will. I don't think there's any need for explicit history for each trigger. One future change I'm planning is to log an activity into the incident when each trigger fires. That might be enough as a basic history for the user.